pkg/minishift/clusterup/clusterup.go

```diff
@@ -0,0 +1,227 @@
+limitations under the License.
+*/
+
+package clusterup
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/blang/semver"
+	"github.com/docker/machine/libmachine/host"
+	"github.com/docker/machine/libmachine/provision"
+
+	"github.com/minishift/minishift/pkg/minikube/constants"
+	"github.com/minishift/minishift/pkg/minikube/kubeconfig"
+	"github.com/minishift/minishift/pkg/minishift/addon/command"
+	"github.com/minishift/minishift/pkg/minishift/addon/manager"
+	"github.com/minishift/minishift/pkg/minishift/oc"
+	"github.com/minishift/minishift/pkg/util"
+	"github.com/minishift/minishift/pkg/version"
+	"os"
+	"strings"
+	"time"
+)
+
+const (
+	ip_key             = "ip"
+	routing_suffix_key = "routing-suffix"
+)
+
+type ClusterUpConfig struct {
+	OpenShiftVersion string
+	MachineName      string
+	Ip               string
+	Port             int
+	RoutingSuffix    string
+	HostPvDir        string
+	User             string
+	Project          string
+	KubeConfigPath   string
+	OcPath           string
+}
+
+// ClusterUp downloads and installs the oc binary in order to run 'cluster up'
+func ClusterUp(config *ClusterUpConfig, clusterUpParams map[string]string, runner util.Runner) error {
+	cmdArgs := []string{"cluster", "up", "--use-existing-config"}
+
+	for key, value := range clusterUpParams {
+		if !oc.SupportFlag(key, config.OcPath, runner) {
+			errors.New(fmt.Sprintf("Flag %s is not supported for oc version %s. Use 'openshift-version' flag to select a different version of OpenShift.", key, config.OpenShiftVersion))
+		}
+		cmdArgs = append(cmdArgs, "--"+key)
+		cmdArgs = append(cmdArgs, value)
+	}
+
+	exitCode := runner.Run(os.Stdout, os.Stderr, config.OcPath, cmdArgs...)
+	if exitCode != 0 {
+		errors.New("Error starting the cluster.")
+	}
+	return nil
+}
+
+// PostClusterUp runs the Minishift specific provisioning after 'cluster up' has run
+func PostClusterUp(clusterUpConfig *ClusterUpConfig, sshCommander provision.SSHCommander, addOnManager *manager.AddOnManager) error {
+	err := kubeconfig.CacheSystemAdminEntries(clusterUpConfig.KubeConfigPath, getConfigClusterName(clusterUpConfig.Ip, clusterUpConfig.Port))
+	if err != nil {
+		return err
+	}
+
+	ocRunner, err := oc.NewOcRunner(clusterUpConfig.OcPath, clusterUpConfig.KubeConfigPath)
+	if err != nil {
+		return err
+	}
+
+	err = ocRunner.AddSudoerRoleForUser(clusterUpConfig.User)
+	if err != nil {
+		return err
+	}
+
+	err = ocRunner.AddCliContext(clusterUpConfig.MachineName, clusterUpConfig.Ip, clusterUpConfig.User, clusterUpConfig.Project)
+	if err != nil {
+		return err
+	}
+
+	err = configurePersistentVolumes(clusterUpConfig.HostPvDir, addOnManager, sshCommander, ocRunner)
+	if err != nil {
+		return err
+	}
+
+	err = applyAddOns(addOnManager, clusterUpConfig.Ip, clusterUpConfig.RoutingSuffix, ocRunner, sshCommander)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+//EnsureHostDirectoriesExist Ensures the specified directories exist on the virtual machine and creates them if not
```

Comment:
Space missing in between `//` and `EnsureHostDirectoriesExist`

Also, comment should be
```
// EnsureHostDirectoriesExist ensures that the specified directories exist on the VM and creates them if does not exist
```